pypesto/visualize/optimizer_convergence.py

```diff
@@ -0,0 +1,58 @@
+import matplotlib.pyplot as plt
+import pandas as pd
+import seaborn as sns
+import numpy as np
+
+
+def optimizer_convergence(result,
+                          ax=None,
+                          size=(18.5, 10.5)):
+    """
+    Scatter plot of function values and gradient values at the end of
+    optimization. Optimizer exit-message is encoded in color. Can help
+    identifying convergence issues in optimization and guide tolerance
+    refinement etc.
+
+    Parameters
+    ----------
+
+    result: pypesto.Result
+        Optimization result obtained by 'optimize.py'
+
+    ax: matplotlib.Axes, optional
+        Axes object to use.
+
+    size: tuple, optional
+        Figure size (width, height) in inches. Is only applied when no ax
+        object is specified
+
+    Returns
+    -------
+
+    ax: matplotlib.Axes
+        The plot axes.
+    """
+    if ax is None:
+        ax = plt.subplots()[1]
+        fig = plt.gcf()
+        fig.set_size_inches(*size)
+
+    fvals = result.optimize_result.get_for_key('fval')
+    # ideally use problem.get_reduced_vector here, but checking for nans
```

Comment:
why not use get_reduced vector?

Reply:
see comment, would require passing problem as argument here.

Reply:
Should always hold a result (/result info in the future) https://github.com/ICB-DCM/pyPESTO/blob/master/pypesto/result.py#L201, but fine as is.